integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala

```diff
@@ -121,10 +124,17 @@ object Compactor {
           segmentIdToLoadStartTimeMapping(validSegments.head),
           SegmentStatus.SUCCESS,
           carbonLoadModelForMergeDataFiles.getFactTimeStamp, rebuiltSegments.toList.asJava)
-
+        siCompactionIndexList ::= indexCarbonTable
       } catch {
         case ex: Exception =>
           LOGGER.error(s"Compaction failed for SI table ${secondaryIndex.indexName}", ex)
+          siCompactionIndexList.foreach { indexCarbonTable =>
+            sparkSession.sql(
+              s"""ALTER TABLE ${carbonLoadModel.getDatabaseName}.${
```

Comment:
please correct the format for sparksession.sql(""), please check the other files where we do alter set for SI, can follow the same

Reply:
OK